src/main/java/roomescape/controller/ReservationApiController.java

```diff
@@ -15,54 +13,59 @@
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 import roomescape.domain.Reservation;
+import roomescape.dto.request.ReservationRequestDto;
+import roomescape.dto.response.ReservationResponseDto;
 import roomescape.exception.ErrorMessage;
 import roomescape.exception.ReservationNotFoundException;
+import roomescape.service.ReservationDto;
+import roomescape.service.ReservationService;
 
-@RestController // @Controller + @ResponseBody 특정 클래스가 RESTful 웹 서비스의 컨트롤러 역할을 하도록 지정한다
+@RestController // @Controller + @ResponseBody: 특정 클래스가 RESTful 웹 서비스의 컨트롤러 역할을 하도록 지정
 public class ReservationApiController {
 
-  private List<Reservation> reservations = new ArrayList<>();
-  private AtomicLong index = new AtomicLong(1);
+  private final ReservationService reservationService;
+
+  public ReservationApiController(ReservationService reservationService) {
+    this.reservationService = reservationService;
+  }
 
   @GetMapping("/reservations")
-  public ResponseEntity<List<Reservation>> readReservation() {
-    return ResponseEntity.ok(reservations);
-    // GET /reservations 요청이 들어왔을 때 저장된 모든 예약 정보를 반환한다
-    // ResponseEntity 는 HTTP 응답을 위한 클래스이다. ok() 메소드는 200 OK 상태 코드를 반환한다
-    // body() 메서드는 응답 본문에 들어갈 데이터를 설정한다. 반환할 데이터는 reservations 필드에 저장된 예약 정보이다
+  public ResponseEntity<List<ReservationResponseDto>> readReservation() {
+    List<ReservationDto> reservations = reservationService.findAll();
+    List<ReservationResponseDto> response =
+        reservations.stream()
+                    .map(ReservationResponseDto::from)
+                    .collect(Collectors.toList());
```

Comment:
```suggestion
    List<ReservationResponseDto> response =
        reservations.stream()
                    .map(ReservationResponseDto::from)
                    .toList();
```
stream을 사용해 list로 만드는 방법으로 `.collect(Collectors.toList())`, `.collect(Collectors.toUnmodifiableList())`, `.toList()`가 있는데 이 셋의 차이는 무엇일까요? 어떤 방법이 더 좋을까요?

Reply:
요거 확인 못하신 것 같아요 🥲

Reply:
`.collect(Collectors.toList())`, `.collect(Collectors.toUnmodifiableList())`
두 리스트 반환 방식은 불변성을 보장하고 불변성을 명시적으로 보이기 위함의 차이가 있습니다.
현재 상황에서는 추가적인 작업이 없기 때문에 불변성을 명시적으로 보일 필요가 없습니다.

`toList()`로 반영했습니다.